system-addon/lib/HighlightsFeed.jsm

```diff
@@ -54,6 +55,12 @@ this.HighlightsFeed = class HighlightsFeed {
   }
 
   async fetchHighlights(broadcast = false) {
+    if (this._updateInProgress) {
```

Comment:
The only issue I've seen with this approach is if an update is in progress where an actual change happened (e.g. something was edited, blocked) the old data is returned instead of the more recent data

Reply:
@k88hudson  We could add `&& !broadcast` if that's the only issue. What about removing refresh entirely from new tab load and defer to SYSTEM_TICK?